builtin/providers/aws/resource_aws_route53_record.go

```diff
@@ -94,6 +94,35 @@ func resourceAwsRoute53Record() *schema.Resource {
 				Optional: true,
 			},
 
+			"region": &schema.Schema{ // AWS region from which to evaluate latency
```

Comment:
The usual convention is to have the Terraform properties match with the AWS API or UI labels for things, but this name seems rather confusing and I think you agree since you put a comment on it trying to elaborate. :grinning:

Perhaps this should be called ``latency_routing_region``. My intent with this name is to indicate that we're giving the region that will be used for latency routing, and not (as would normally be the case where "region" is used elsewhere in Terraform) where this record resource will be hosted.

Alternatively, riffing off how this is arranged in the Route53 UI, I might consider having a ``routing_policy`` child block that contains all of these routing policy options. This is similar to how it's placed in a separate box in the Route53 console, distinct from the TTL and value.

```js
resource "aws_route53_record" "foo" {
    zone_id = "${aws_route53_zone.primary.zone_id}"
    name = "www.example.com"
    type = "A"
    ttl = "300"
    records = ["${aws_eip.lb.public_ip}"]

    routing_policy {
        region = "us-west-2"
    }
}
```

Then at least it's a bit clearer that this region is routing-related. Your other set of mutually-exclusive fields 

That last alternative still misses the "routing policy" field from the UI where you choose a type of routing, indicating that this is latency-based routing, so maybe this is better:

```js
resource "aws_route53_record" "foo" {
    zone_id = "${aws_route53_zone.primary.zone_id}"
    name = "www.example.com"
    type = "A"
    ttl = "300"
    records = ["${aws_eip.lb.public_ip}"]

    latency_routing_policy {
        region = "us-west-2"
    }
}
```

...and then there could be other mutually-exclusive child blocks for ``weighted_routing_policy``, ``failover_routing_policy`` and ``geolocation_routing_policy``, each containing the set of fields that gets displayed when you select the corresponding option for routing policy in the Route53 console.

Not sure which of these options I like the best. I think I lean towards the latter since I think it's most user-friendly to prefer to mimic the AWS UI rather than the API when they are in conflict, but YMMV.